File: scripts/svg_to_vector.js
const fs = require('fs');
const path = require('path');

const root = path.resolve(__dirname, '..');
const drawableDir = path.join(root, 'app', 'src', 'main', 'res', 'drawable');

function parsePathTag(tag) {
  const dMatch = tag.match(/\bd="([^"]+)"/);
  const fillMatch = tag.match(/(?<![-\w])fill="([^"]+)"/);
  if (!dMatch) return null;
  return { d: dMatch[1], fill: fillMatch?.[1] ?? '#394149' };
}

function parseTranslate(transform) {
  const match = transform.match(/translate\(\s*([^,\s]+)\s*,\s*([^)]+)\)/);
  if (!match) return null;
  return { x: parseFloat(match[1]), y: parseFloat(match[2]) };
}

function extractContent(svgPath) {
  const text = fs.readFileSync(svgPath, 'utf8');
  const viewBoxMatch = text.match(/viewBox="([^"]+)"/);
  const viewBox = viewBoxMatch[1].split(/\s+/);
  const nodes = [];

  const groupRegex = /<g[^>]*id="([^"]+)"[^>]*transform="([^"]*)"[^>]*>([\s\S]*?)<\/g>/g;
  let groupMatch;
  while ((groupMatch = groupRegex.exec(text)) !== null) {
    const translate = parseTranslate(groupMatch[2]);
    const pathRegex = /<path[^>]*\/?>/g;
    let pathMatch;
    while ((pathMatch = pathRegex.exec(groupMatch[3])) !== null) {
      const parsed = parsePathTag(pathMatch[0]);
      if (parsed) {
        nodes.push({ ...parsed, translate });
      }
    }
  }

  if (nodes.length === 0) {
    const pathRegex = /<path[^>]*\/?>/g;
    let pathMatch;
    while ((pathMatch = pathRegex.exec(text)) !== null) {
      const parsed = parsePathTag(pathMatch[0]);
      if (parsed) nodes.push({ ...parsed, translate: null });
    }
  }

  return { viewBox, nodes };
}

function writeVector(name) {
  const { viewBox, nodes } = extractContent(path.join(root, `${name}.svg`));
  const width = parseFloat(viewBox[2]);
  const height = parseFloat(viewBox[3]);
  const lines = [
    '<?xml version="1.0" encoding="utf-8"?>',
    '<vector xmlns:android="http://schemas.android.com/apk/res/android"',
    `    android:width="${width}dp"`,
    `    android:height="${height}dp"`,
    `    android:viewportWidth="${width}"`,
    `    android:viewportHeight="${height}">`,
  ];

  for (const node of nodes) {
    const fill = node.fill ?? '#394149';
    if (node.translate) {
      lines.push('    <group');
      lines.push(`        android:translateX="${node.translate.x}"`);
      lines.push(`        android:translateY="${node.translate.y}">`);
    }
    lines.push('        <path');
    lines.push(`            android:fillColor="${fill}"`);
    lines.push('            android:fillType="evenOdd"');
    lines.push(`            android:pathData="${node.d}" />`);
    if (node.translate) {
      lines.push('    </group>');
    }
  }

  lines.push('</vector>');
  const output = path.join(drawableDir, `${name}.xml`);
  fs.writeFileSync(output, lines.join('\n') + '\n', 'utf8');
  console.log(`Wrote ${output} (${nodes.length} paths)`);
}

for (const logo of ['logo_icon', 'logo_full']) {
  writeVector(logo);
}

File: scripts/svg_to_vector.py
import re
import pathlib

ROOT = pathlib.Path(__file__).resolve().parents[1]
DRAWABLE = ROOT / "app" / "src" / "main" / "res" / "drawable"


def extract_paths(svg_path: pathlib.Path):
    text = svg_path.read_text(encoding="utf-8")
    viewbox = re.search(r'viewBox="([^"]+)"', text).group(1).split()
    paths = []
    for match in re.finditer(r'<path[^>]*d="([^"]+)"[^>]*fill="([^"]+)"', text):
        paths.append((match.group(1), match.group(2)))
    return viewbox, paths


def write_vector(name: str):
    svg_path = ROOT / f"{name}.svg"
    viewbox, paths = extract_paths(svg_path)
    width = float(viewbox[2])
    height = float(viewbox[3])
    lines = [
        '<?xml version="1.0" encoding="utf-8"?>',
        '<vector xmlns:android="http://schemas.android.com/apk/res/android"',
        f'    android:width="{width}dp"',
        f'    android:height="{height}dp"',
        f'    android:viewportWidth="{width}"',
        f'    android:viewportHeight="{height}">',
    ]
    for path_data, fill in paths:
        lines.extend(
            [
                "    <path",
                f'        android:fillColor="{fill}"',
                '        android:fillType="evenOdd"',
                f'        android:pathData="{path_data}" />',
            ]
        )
    lines.append("</vector>")
    output = DRAWABLE / f"{name}.xml"
    output.write_text("\n".join(lines) + "\n", encoding="utf-8")
    print(f"Wrote {output} ({len(paths)} paths, {output.stat().st_size} bytes)")


if __name__ == "__main__":
    for logo in ("logo_icon", "logo_full"):
        write_vector(logo)
